File: pdlearning/src/microservices/newsfeed-microservice/Api/Application/Models/CourseFeedModel.cs
using System;
using Microservice.NewsFeed.Domain.Entities;
using Microservice.NewsFeed.Domain.ValueObject;

namespace Microservice.NewsFeed.Application.Models
{
    public class CourseFeedModel
    {
        public CourseFeedModel(PdpmSuggestCourseFeed courseFeed)
        {
            Id = courseFeed.Id;
            CourseId = courseFeed.CourseId;
            UserId = courseFeed.UserId;
            CreatedDate = courseFeed.CreatedDate;
            ThumbnailUrl = courseFeed.ThumbnailUrl;
            CourseName = courseFeed.CourseName;
            UpdateInfo = courseFeed.UpdateInfo;
            Url = courseFeed.Url;
        }

        public Guid Id { get; set; }

        public Guid CourseId { get; set; }

        public Guid UserId { get; set; }

        public DateTime CreatedDate { get; set; }

        public string ThumbnailUrl { get; set; }

        public string CourseName { get; set; }

        public UpdateCourseInfoType UpdateInfo { get; set; }

        public string Url { get; set; }

        public string Type => nameof(PdpmSuggestCourseFeed);
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/newsfeed/models/course-feed.model.ts
import { INewsfeedBaseModel, NewsFeedUpdateInfo, NewsfeedType } from './newsfeed.model';

export interface ICourseFeedModel extends INewsfeedBaseModel {
  courseId: string;
  userId: string;
  createdDate: Date;
  thumbnailUrl: string;
  courseName: string;
  updateInfo: NewsFeedUpdateInfo;
  url: string;
}

export class CourseFeedModel implements ICourseFeedModel {
  public id: string;
  public type: NewsfeedType;
  public courseId: string;
  public userId: string;
  public createdDate: Date;
  public thumbnailUrl: string;
  public courseName: string;
  public updateInfo: NewsFeedUpdateInfo;
  public url: string;
  constructor(data?: ICourseFeedModel) {
    if (data == null) {
      return;
    }
    this.id = data.id;
    this.type = data.type;
    this.courseId = data.courseId;
    this.userId = data.userId;
    this.createdDate = new Date(data.createdDate);
    this.thumbnailUrl = data.thumbnailUrl;
    this.courseName = data.courseName;
    this.updateInfo = data.updateInfo;
    this.url = data.url;
  }
}
